components/measure-upload/EvaluationServiceVerifier.tsx

```diff
@@ -0,0 +1,123 @@
+import { Alert, Button, Group, Space, TextInput } from '@mantine/core';
+import React, { useEffect, useState } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import { evaluationState } from '../../state/atoms/evaluation';
+import { measureBundleState } from '../../state/atoms/measureBundle';
+import { showNotification } from '@mantine/notifications';
+import { IconAlertCircle, IconCircleCheck } from '@tabler/icons';
+
+export default function EvaluationServiceVerifier() {
+  const [isLoadingId, setIsLoadingId] = useState(false);
+  const { content } = useRecoilValue(measureBundleState);
+  const [{ evaluationServiceUrl, evaluationMeasureId }, setEvaluation] = useRecoilState(evaluationState);
+
+  useEffect(() => {
+    // Called when the measure content value changes
+    setEvaluation(e => ({
+      ...e,
+      evaluationMeasureId: ''
+    }));
+  }, [content, setEvaluation]);
+
+  /**
+   * Handles updating of the Evaluation Service url field and associated state variables
+   */
+  function handleUrlChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setEvaluation({
+      evaluationServiceUrl: event.currentTarget.value,
+      evaluationMeasureId: ''
+    });
+  }
+
+  /**
+   * Sends a `GET` request to the `/Measure` endpoint of the specified evaluation service URL.
+   * Parses and stores the url and the if of the matching (same url/version) Measure if successful
+   */
+  async function verifyMeasure() {
+    setIsLoadingId(true);
+    let parsedEvaluationUrl = evaluationServiceUrl;
+    if (evaluationServiceUrl.slice(-1) === '/') {
+      parsedEvaluationUrl = evaluationServiceUrl.slice(0, -1);
+    }
+    try {
+      const response = await fetch(`${parsedEvaluationUrl}/Measure`);
+      const responseBody = await response.json();
+      if (response.status === 200) {
+        if (responseBody.entry.length === 0) {
+          showNotification({
+            icon: <IconAlertCircle />,
+            title: 'No measures found',
+            message: `Evalution service hosted at ${parsedEvaluationUrl} returned an empty 
+            FHIR Bundle when queried for Measures. Ensure Measures are present on server.`,
+            color: 'red'
+          });
+          setIsLoadingId(false);
+          return;
+        }
+        const selectedMeasure = content?.entry?.find(e => e.resource?.resourceType === 'Measure')
```

Comment:
Not a necessary change but could consider having a notification if there is no measure present:
```suggestion
const selectedMeasure = content?.entry?.find(e => e.resource?.resourceType === 'Measure')
if (!selectedMeasure) {
  showNotification({
    icon: <IconAlertCircle />,
    title: 'No Measure selected',
    message: 'No Measure resource found in the loaded bundle. Please ensure a Measure is loaded.',
    color: 'red'
  });
  setIsLoadingId(false);
  return;
}
```

Reply:
yea I didnt notice this at first was just looking at the code. In that case I dont think we need the notification and its good as is. 

completely unnecessary but  would it make sense to have all of `Step 2` "greyed out" until a measure has been entered? this could help with the flow, but on the other had I could see this getting annoying if someone has the URL copied at first and then wants to enter in the measure after. I know this is an internal tool but could help with user readability for the future. thoughts?

Reply:
Agree that there's not a perfect user flow here, since Step 1 implies strong preference to come before Step 2. I don't think it's necessary to completely disable Step 2 since there's not a great reason to disallow interacting with those components. There could be a better overall flow here, but I think that's out of scope for this feature.